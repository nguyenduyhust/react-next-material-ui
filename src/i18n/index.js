const NextI18Next = require('next-i18next').default;
const path = require('path');

module.exports = new NextI18Next({
  defaultLanguage: 'en',
  otherLanguages: ['ja'],
  browserLanguageDetection: true,
  serverLanguageDetection: true,
  detection: {
    order: ['querystring', 'cookie', 'header', 'navigator'],
    lookupQuerystring: 'lng',
    lookupCookie: 'lng',
    lookupHeader: 'accept-language',
    caches: ['cookie'],
  },
  localePath: path.resolve('public/static/locales'),
});
