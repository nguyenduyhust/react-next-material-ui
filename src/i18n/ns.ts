export enum NamespaceEnum {
  COMMON = 'common',
  HOME_PAGE = 'homepage',
}
export const ns: string | Array<string> = [NamespaceEnum.COMMON, NamespaceEnum.HOME_PAGE];
export const defaultNS: string = NamespaceEnum.COMMON;
