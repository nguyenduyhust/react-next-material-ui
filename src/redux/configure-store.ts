import { createStore, combineReducers, applyMiddleware } from 'redux';
import promiseMiddleware from 'redux-promise';
import {
  typePendingReducerSet,
  createTypeReduxInitialState,
  typeReduxMiddleware,
} from './type-redux';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';

import * as AppReducer from './reducers/app-reducer';

export const rootReducer = combineReducers({
  ...typePendingReducerSet,
  appState: AppReducer.reducer,
});

export interface RootState {
  appState: AppReducer.State;
}

export const InitialState: RootState = Object.assign(createTypeReduxInitialState(), {
  appState: AppReducer.initialState,
});

const middlewares: any[] = [typeReduxMiddleware, promiseMiddleware];

export const configureStore = (initialState?: RootState) => {
  return createStore(
    rootReducer,
    { ...InitialState, ...initialState },
    composeWithDevTools(applyMiddleware(...middlewares)),
  );
};
