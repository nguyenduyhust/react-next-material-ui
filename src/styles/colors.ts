export const BLACK_COLOR = '#000000';
export const WHITE_COLOR = '#FFFFFF';
export const TORY_BLUE_COLOR = '#344090';
export const NAVY_BLUE_COLOR = '#1976D2';
export const PATTENS_BLUE_COLOR = '#D8DFE5';
export const MID_NIGHT_COLOR = '#233044';
export const ALICE_BLUE_COLOR = '#F7F9FC';
export const NOBEL_COLOR = '#9E9E9E';
export const CHARCOAL_COLOR = '#424242';
