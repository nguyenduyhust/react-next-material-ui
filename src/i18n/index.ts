import NextI18Next from 'next-i18next';
import path from 'path';
import moment from 'moment';
import { ns, defaultNS } from './ns';

const {
  Trans,
  Link,
  Router,
  i18n,
  initPromise,
  config,
  useTranslation,
  withTranslation,
  appWithTranslation,
} = new NextI18Next({
  ns,
  defaultNS,
  defaultLanguage: 'en',
  otherLanguages: ['ja'],
  browserLanguageDetection: true,
  serverLanguageDetection: true,
  detection: {
    order: ['querystring', 'cookie', 'header', 'navigator'],
    lookupQuerystring: 'lng',
    lookupCookie: 'lng',
    lookupHeader: 'accept-language',
    caches: ['cookie'],
  },
  localePath: path.resolve('public/static/locales'),
  interpolation: {
    format: function (value, format, lng) {
      if (value instanceof Date) {
        return moment(value).format(format);
      }
      return value;
    },
  },
});

export {
  Trans,
  Link,
  Router,
  i18n,
  initPromise,
  config,
  useTranslation,
  withTranslation,
  appWithTranslation,
};
export { NamespaceEnum } from './ns';
