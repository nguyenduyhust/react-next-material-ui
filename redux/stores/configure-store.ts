import { createStore, combineReducers, applyMiddleware } from 'redux';
import promiseMiddleware from 'redux-promise';
import { typePendingReducerSet, createTypeReduxInitialState, typeReduxMiddleware } from 'type-redux';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';

import * as AppReducer from './app-reducer';

export const rootReducer = combineReducers({
  ...typePendingReducerSet,
  appState: AppReducer.reducer,
});

export interface Store {
  appState: AppReducer.State;
}

export const InitialState: Store = Object.assign(createTypeReduxInitialState(), {
  appState: AppReducer.initialState,
});

const middlewares: any[] = [typeReduxMiddleware, promiseMiddleware];

export const configureStore = (initialState?: Store) => {
  return createStore(rootReducer, { ...InitialState, ...initialState }, composeWithDevTools(applyMiddleware(...middlewares)));
};
