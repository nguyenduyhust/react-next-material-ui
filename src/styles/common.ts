import { CSSProperties } from '@material-ui/styles';

export const link: CSSProperties = {
  textDecoration: 'none',
  '&:hover': {
    textDecoration: 'underline',
  },
};
