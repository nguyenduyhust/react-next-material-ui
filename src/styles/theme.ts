import { createMuiTheme, Theme } from '@material-ui/core/styles';
import {
  BLACK_COLOR,
  WHITE_COLOR,
  TORY_BLUE_COLOR,
  NAVY_BLUE_COLOR,
  PATTENS_BLUE_COLOR,
  MID_NIGHT_COLOR,
  ALICE_BLUE_COLOR,
  NOBEL_COLOR,
  CHARCOAL_COLOR,
} from '../constants/colors';
import {
  XS_BREAKPOINT,
  SM_BREAKPOINT,
  MD_BREAKPOINT,
  LG_BREAKPOINT,
  XL_BREAKPOINT,
} from '../constants/breakpoints';

const theme: Theme = createMuiTheme({
  colors: {
    black: BLACK_COLOR,
    white: WHITE_COLOR,
    toryBlue: TORY_BLUE_COLOR,
    navyBlue: NAVY_BLUE_COLOR,
    pattensBlue: PATTENS_BLUE_COLOR,
    midnight: MID_NIGHT_COLOR,
    aliceBlue: ALICE_BLUE_COLOR,
    nobel: NOBEL_COLOR,
    charcoal: CHARCOAL_COLOR,
  },
  breakpoints: {
    values: {
      xs: XS_BREAKPOINT,
      sm: SM_BREAKPOINT,
      md: MD_BREAKPOINT,
      lg: LG_BREAKPOINT,
      xl: XL_BREAKPOINT,
    },
  },
  spacing: 8,
  palette: {
    primary: {
      main: NAVY_BLUE_COLOR,
    },
  },
});

export default theme;
