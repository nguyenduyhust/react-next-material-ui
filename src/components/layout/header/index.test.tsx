// libs
import React from 'react';
import { mount } from 'enzyme';
// providers
import { I18nextProvider } from 'react-i18next';
import { SnackbarProvider } from 'notistack';
import { Provider } from 'react-redux';
import { ThemeProvider } from '@material-ui/core/styles';
// redux
import { configureStore, InitialState } from '~/redux/configure-store';
// i18n
import { i18n } from '~/i18n';
// theme
import theme from '~/styles/theme';
// components
import Component from '.';
// mocks
import firebase from 'firebase';
import { FirebaseService } from '~/services/firebase.service';
jest.spyOn<any, any>(firebase, 'auth').mockImplementation(() => {
  return {
    onAuthStateChanged: jest.fn(),
    currentUser: {
      displayName: 'testDisplayName',
      email: 'test@test.com',
      emailVerified: true,
    },
  };
});
jest.spyOn<any, any>(FirebaseService, 'snapshotUserPreference').mockImplementation(() => {
  return {};
});

describe('Component test', () => {
  test('Renders correctly', () => {
    const store = configureStore({
      ...InitialState,
      appState: {
        ...InitialState.appState,
        profile: {
          ...InitialState.appState.profile,
          user: {
            uid: 'uid',
            email: 'test@test.com',
            displayName: 'Test',
            photoURL: 'http://test.com',
          } as firebase.User,
        },
      },
    });
    const tree = mount(
      <SnackbarProvider>
        <I18nextProvider i18n={i18n}>
          <ThemeProvider theme={theme}>
            <Provider store={store}>
              <Component />
            </Provider>
          </ThemeProvider>
        </I18nextProvider>
      </SnackbarProvider>,
    ).html();
    expect(tree).toMatchSnapshot();
  });
});
