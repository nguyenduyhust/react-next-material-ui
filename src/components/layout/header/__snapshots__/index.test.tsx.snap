// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component test Renders correctly 1`] = `"<header class=\\"MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-1 mui-fixed MuiPaper-elevation0\\"><div class=\\"MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-2 MuiToolbar-gutters\\"><div class=\\"makeStyles-root-10\\"><div class=\\"makeStyles-searchIcon-11\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z\\"></path></svg></div><div class=\\"MuiInputBase-root makeStyles-inputRoot-12\\"><input placeholder=\\"header.search_box_placeholder\\" type=\\"text\\" class=\\"MuiInputBase-input makeStyles-inputInput-13\\" value=\\"\\"></div></div><div class=\\"makeStyles-grow-8\\"></div><div class=\\"makeStyles-accountInfo-6\\"><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\"><img class=\\"makeStyles-avatar-9\\" src=\\"http://test.com\\" alt=\\"\\"></span><span class=\\"MuiTouchRipple-root\\"></span></button></div></div></header>"`;
